lib/internal/test_runner/runner.js

```diff
@@ -248,14 +220,79 @@ class FileTest extends Test {
       this.#buffer = [];
     }
   }
-  addToReport(ast) {
-    this.#accumulateReportItem(ast);
+  addToReport(item) {
+    this.#accumulateReportItem(item);
     if (!this.isClearToSend()) {
-      ArrayPrototypePush(this.#buffer, ast);
+      ArrayPrototypePush(this.#buffer, item);
       return;
     }
     this.#drainBuffer();
-    this.#handleReportItem(ast);
+    this.#handleReportItem(item);
+  }
+  parseMessage(readData) {
+    if (readData.length === 0) return;
+
+    ArrayPrototypePush(this.#messageBuffer, readData);
+    this.#messageBufferSize += readData.length;
+
+    // Index 0 should always be present because we just pushed data into it.
+    let messageBufferHead = this.#messageBuffer[0];
+    let headerIndex = messageBufferHead.indexOf(v8Header);
```

Comment:
```suggestion
    let headerIndex = ArrayPrototypeIndexOf(messageBufferHead, v8Header);
```

Reply:
messageBufferHead is a Buffer

Reply:
If it's a Buffer, you should be able to use ``Uint8ArrayPrototypeIndexOf``

Reply:
I have tried but:
```js
./node --expose-internals -p "require('internal/test/binding').primordials.Uint8ArrayPrototypeIndexOf()" 
```
results in `require(...).primordials.Uint8ArrayPrototypeIndexOf is not a function`
and ```js
./node --expose-internals -p "require('internal/test/binding').primordials.TypedArrayPrototypeIndexOf(Buffer.from('Hello'), Buffer.from('ll'))"
```

Reply:
Try `TypedArrayPrototypeIndexOf`